relayer/naive-strategy.go

```diff
@@ -232,6 +238,36 @@ func (nrs *NaiveStrategy) UnrelayedAcknowledgements(src, dst *Chain) (*RelaySequ
 
 // HandleEvents defines how the relayer will handle block and transaction events as they are emitted
 func (nrs *NaiveStrategy) HandleEvents(src, dst *Chain, events map[string][]string) {
+	// check for misbehaviour
+	if hdrs, ok := events[fmt.Sprintf("%s.%s", updateCliTag, headerTag)]; ok {
+		for i, hdr := range hdrs {
+			clientIDs := events[fmt.Sprintf("%s.%s", updateCliTag, clientIDTag)]
+
+			if src.PathEnd.ClientID == clientIDs[i] && dst.PathEnd.ClientID == clientIDs[i] {
+				hdrBytes, err := hex.DecodeString(hdr)
+				if err == nil {
+					exportedHeader, err := clienttypes.UnmarshalHeader(src.Encoding.Marshaler, hdrBytes)
+					if err == nil {
+						gotHeader, ok := exportedHeader.(*tmclient.Header)
+						if ok {
+							trustedHeader, err := src.GetLatestLightHeader()
```

Comment:
We need to compare [consensus states](https://github.com/cosmos/ibc-go/blob/333c1f338b2a14a1928a6f8ab64c37123c0e97b6/modules/light-clients/07-tendermint/types/header.go#L18). We should check the consensus state of the emitted header against a header which we construct/retrieve from our light client.

You'll likely need to use this [function](https://github.com/cosmos/relayer/blob/master/relayer/ibc-client.go#L112) and passing in the height of the emitted header. Then use the reflect DeepEqual check against the consensus states